File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/statx/max_milli.go
package statx

// MaxMilli is the max_milli kernel.
func MaxMilli(series []int64) int64 {
	if len(series) == 0 {
		return 0
	}
	best := series[0]
	for _, value := range series[1:] {
		if value > best {
			best = value
		}
	}
	return best * 1000
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/stats/max_milli.cpp
#include "freight/stats.h"

#include <algorithm>

namespace freight {

// max_milli kernel.
long long stat_max_milli(const std::vector<long long>& series) {
  if (series.empty()) {
    return 0;
  }
  long long best = series[0];
  for (size_t i = 1; i < series.size(); ++i) {
    if (series[i] > best) {
      best = series[i];
    }
  }
  return best * 1000;
}

}  // namespace freight
